packages/astro/src/content/vite-plugin-content-virtual-mod.ts

```diff
@@ -72,6 +76,29 @@ export function astroContentVirtualModPlugin({
 			if (id === DATA_STORE_VIRTUAL_ID) {
 				return RESOLVED_DATA_STORE_VIRTUAL_ID;
 			}
+
+			if (hasContentModuleFlag(id)) {
+				const [, query] = id.split('?');
+				const params = new URLSearchParams(query);
+				const importerParam = params.get('importer');
+				const importerPath = importerParam
+					? fileURLToPath(new URL(importerParam, settings.config.root))
+					: importer;
```

Comment:
This is resolving the wrong thing. The `importer` arg passed to `resolveId` isn't the specifier: it's the file that is doing the import.